src/main/java/amazin/security/AuthoritiesConstants.java

```diff
@@ -0,0 +1,13 @@
+package amazin.security;
+
+public final class AuthoritiesConstants {
+
+    public static final String ADMIN = "ROLE_ADMIN";
+
+    public static final String USER = "ROLE_USER";
+
+    public static final String ANONYMOUS = "ROLE_ANONYMOUS";
+
+    private AuthoritiesConstants() {
```

Comment:
this constuctor is private, which is not the default (public). Is this needed by some library?

Reply:
defined as private since this is just constants, other spring security examples I found online are doing the same thing

Reply:
then it is fine

Reply:
> defined as private since this is just constants, other spring security examples I found online are doing the same thing

Is this class then acting as a global variable? is this best practice?

Reply:
Everything in Java has to be a class.